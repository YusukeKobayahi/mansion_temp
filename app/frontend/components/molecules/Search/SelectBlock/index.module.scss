@import "~/styles/variables";
@import "~/styles/utils";

.label {
  font-size: 12px;
  font-weight: bold;
  color: #373a3c;
  text-align: left;
  padding-right: 17px;
}

.unit {
  font-weight: lighter;
  color: #646464;
}

.content {
  font-size: 12px;
  align-items: center;
  display: grid;
  grid-template-columns: 7fr 9fr 3fr;
  column-gap: 7px;
}
