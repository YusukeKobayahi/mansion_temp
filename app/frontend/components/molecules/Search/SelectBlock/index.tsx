import styles from "./index.module.scss";
import { ReactNode } from "react";

import BasicBlock from "~/components/molecules/Search/BasicBlock";

type SelectBlockProps = {
  labelText: string;
  children: ReactNode;
  unitText: string;
  sectionBar?: boolean;
};

const SelectBlock: React.FC<SelectBlockProps> = ({
  labelText,
  children,
  unitText,
  sectionBar = true,
}: SelectBlockProps) => {
  return (
    <BasicBlock sectionBar={sectionBar}>
      <div className={styles.content}>
        <p className={styles.label}>{labelText}</p>
        <div>{children}</div>
        <p className={styles.unit}>{unitText}</p>
      </div>
    </BasicBlock>
  );
};

export default SelectBlock;
