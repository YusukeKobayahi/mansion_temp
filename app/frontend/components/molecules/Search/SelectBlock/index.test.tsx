import { render } from "@testing-library/react";
import SelectBlock from ".";

describe("components/molecules/Search/SelectBlock", () => {
  const select = (
    <select>
      <option value="10">10</option>
      <option value="20">20</option>
      <option value="30">30</option>
    </select>
  );
  it("正常に表示される", () => {
    const { asFragment } = render(
      <SelectBlock labelText={"徒歩"} unitText={"分"}>
        {select}
      </SelectBlock>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
