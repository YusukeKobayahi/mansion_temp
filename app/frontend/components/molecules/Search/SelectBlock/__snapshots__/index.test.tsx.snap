// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/molecules/Search/SelectBlock 正常に表示される 1`] = `
<DocumentFragment>
  <div
    class="block sectionBar"
  >
    <div
      class="content"
    >
      <p
        class="label"
      >
        徒歩
      </p>
      <div>
        <select>
          <option
            value="10"
          >
            10
          </option>
          <option
            value="20"
          >
            20
          </option>
          <option
            value="30"
          >
            30
          </option>
        </select>
      </div>
      <p
        class="unit"
      >
        分
      </p>
    </div>
  </div>
</DocumentFragment>
`;
